lib/irb/command/copy.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module IRB
+  module Command
+    class Copy < Base
+      category "Workspace"
+      description "Copy command output to clipboard"
+
+      help_message(<<~HELP)
+        Usage: copy (command)
+      HELP
+
+      def execute(arg)
+        value = irb_context.workspace.binding.eval(arg)
+        output = irb_context.inspect_method.inspect_value(value, colorize: false)
+
+        if clipboard_available?
+          copy_to_clipboard(output)
+        else
+          warn "System clipboard not found"
+        end
+      rescue StandardError => e
+        warn "Error: #{e}"
+      end
+
+      private
+
+      def copy_to_clipboard(text)
+        IO.popen(clipboard_program, 'w') do |io|
+          io.write(text)
+        end
+
+        raise IOError.new("Copying to clipboard failed") unless $? == 0
+
+        puts "Copied to system clipboard"
+      end
+
+      def clipboard_program
+        @clipboard_program = "pbcopy" if executable?("pbcopy")
```

Comment:
Should we also have `return if defined?(@clipboard_program)` so we don't need to reevaluate the presence of the executables?

Reply:
I updated it to be `@clipboard_program ||=`, but it looks like the command is re-instantiated each time so it doesn't matter. I could cache the value elsewhere (say, the class), but that has a subtle bug where changes to `IRB.conf[:CLIPBOARD_PROGRAM]` won't reflect once this has been computed.